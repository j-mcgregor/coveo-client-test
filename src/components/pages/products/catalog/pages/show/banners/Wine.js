/* eslint-disable camelcase */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const BreweryBanner = ({
  wine,
  country,
  regions,
  vintage,
  wine_slug,
  score,
  confidence_index,
  journalist_count,
  closeBanner
}) => (
  <Fragment>
    <div className="columns">
      <div className="column col text-right">
        <i className="fas fa-times" onClick={closeBanner} role="presentation" />
      </div>
    </div>
    <div className="columns">
      <div className="column">
        <h2>{wine}</h2>
      </div>
    </div>
    <div className="columns">
      <div className="column col-6 col-sm-5">
        <div>
          {regions.length ? `${regions[0]}, ` : ''}
          {country}
        </div>
        <small>Vintage: {vintage}</small>
      </div>
      <div className="column col-6">
        <div className="columns">
          <div className="column col text-right">
            Average Score ({journalist_count}):{' '}
          </div>
          <div className="column col-3">{score}%</div>
        </div>
        <div className="columns">
          <div className="column col text-right">Confidence Index: </div>
          <div className="column col-3">{confidence_index}</div>
        </div>
        <div className="columns text-right">
          <div className="column col-9">
            <small>
              <a
                href={`http://maps.google.com/?q=${wine_slug}`}
                target="_blank"
                rel="noopener noreferrer"
              >
                See on map
              </a>
            </small>
          </div>
        </div>
      </div>
    </div>
  </Fragment>
);

BreweryBanner.propTypes = {
  wine: PropTypes.string,
  country: PropTypes.string,
  regions: PropTypes.string,
  vintage: PropTypes.string,
  wine_slug: PropTypes.string,
  score: PropTypes.string,
  confidence_index: PropTypes.string,
  journalist_count: PropTypes.string,
  closeBanner: PropTypes.func
};

BreweryBanner.defaultProps = {
  wine: '',
  country: '',
  regions: '',
  vintage: '',
  wine_slug: '',
  score: '',
  confidence_index: '',
  journalist_count: '',
  closeBanner: () => {}
};

export default BreweryBanner;
