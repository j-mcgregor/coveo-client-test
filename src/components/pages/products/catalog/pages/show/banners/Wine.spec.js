import React from 'react';
import { shallow } from 'enzyme';
import Wine from './Wine';

describe('Wine', () => {
  it('should render', () => {
    const props = {
      wine: 'Coveo Pinot Noir',
      country: 'Canada',
      regions: 'Quebec',
      vintage: '2020',
      wine_slug: 'very-good-wine',
      score: '100',
      confidence_index: 'A++',
      journalist_count: '1000',
      closeBanner: () => {}
    };
    const wrapper = shallow(<Wine {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
