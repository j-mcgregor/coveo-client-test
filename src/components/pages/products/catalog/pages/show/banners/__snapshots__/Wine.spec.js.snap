// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wine should render 1`] = `ShallowWrapper {}`;
